psycop/common/model_training_v2/problem_type/problem_type_base.py

```diff
@@ -41,9 +47,37 @@ def __init__(
     def train(
         self,
         X: PolarsFrame,
-        y: PolarsFrame,
-    ) -> float:
-        ...
+        y: pl.Series,
+    ) -> None:
+        self.pipe.fit(X, y)
+
+    def predict_proba(self, x: PolarsFrame) -> pl.Series:
+        return self.pipe.predict_proba(x)
+
+    def evaluate(self, x: PolarsFrame, y: pl.Series) -> TrainingResult:
+        if isinstance(x, pl.LazyFrame):
+            x = x.collect()
+        y_hat_probs = self.pipe.predict_proba(x)
+
+        df = x.with_columns(y_hat_probs, y)
+        eval_dataset = BinaryEvalDataset(
+            pred_time_uuids="pred_time_uuids",  # need to get this column from somewhere!
+            y_hat_probs=y_hat_probs.name,
```

Comment:
If they are pd.Series, do they have a name attribute?

Reply:
yep! the column name is set as .name